### PR TITLE
build(client): restore property-shared-tree-interop ESM build

### PR DESCRIPTION
**Pull request by jason-ha**
+ declare exports
+ test ESM

### CODE REVIEW

File: experimental/PropertyDDS/packages/property-shared-tree-interop/package.json [hunk 1]
```diff
@@ -10,9 +10,19 @@
 	},
 	"license": "MIT",
 	"author": "Microsoft and contributors",
-	"main": "dist/index.js",
-	"module": "lib/index.js",
-	"types": "dist/index.d.ts",
```
Comment (DLehenbauer) [inline experimental/PropertyDDS/packages/property-shared-tree-interop/package.json:15]: Don't we still want main/types to point customers still using node10 at our public rollup?

File: fluidBuild.config.cjs [hunk 2]
```diff
@@ -243,7 +243,9 @@ module.exports = {
 				"^tools/getkeys",
 			],
 			"npm-package-json-esm": [
-				"packages/dds/tree/package.json", // Policy is incorrect about "module" in package.json
+				// Policy is incorrect about "module" in package.json
+				"experimental/PropertyDDS/packages/property-shared-tree-interop/package.json",
+				"packages/dds/tree/package.json",
```
Comment (DLehenbauer) [inline fluidBuild.config.cjs:248]: Will we need to add an entry for every package we convert to '"type": "module"'?

**Review by DLehenbauer — APPROVED** [review]
Thanks, Jason...  just a couple minor questions for my own understanding.

File: experimental/PropertyDDS/packages/property-shared-tree-interop/package.json (hunk 1 above)
Comment (jason-ha) [reply experimental/PropertyDDS/packages/property-shared-tree-interop/package.json:15]: I was thinking no as this package is quite new. (I think I originally removed it when I found no uses of it as a library and later saw that is its intent.) I can restore main/types.?

Comment (DLehenbauer) [reply experimental/PropertyDDS/packages/property-shared-tree-interop/package.json:15]: I'd leave them in for customers who are integrating Fluid into a pre-existing codebase, which describes most of our 3P partners.

(...but I wouldn't hold off merging this PR for it.)

Comment (jason-ha) [reply experimental/PropertyDDS/packages/property-shared-tree-interop/package.json:15]: I restored main and types when the CI build wasn't triggering.
`module` is not a supported field. When we have exports we should just remove module.

**Pull request merged**